lib/Doctrine/DBAL/SQLParserUtils.php

```diff
@@ -196,16 +204,46 @@ static public function expandListParameters($query, $params, $types)
      * 0 => matched fragment string,
      * 1 => offset of fragment in $statement
      *
-     * @param string $statement
+     * @param string                $statement
+     * @param AbstractPlatform|null $platform The platform to use for unquoting.
+     *
      * @return array
      */
-    static private function getUnquotedStatementFragments($statement)
+    static private function getUnquotedStatementFragments($statement, AbstractPlatform $platform = null)
     {
-        $literal = self::ESCAPED_SINGLE_QUOTED_TEXT . '|' .
-                   self::ESCAPED_DOUBLE_QUOTED_TEXT . '|' .
-                   self::ESCAPED_BACKTICK_QUOTED_TEXT . '|' .
-                   self::ESCAPED_BRACKET_QUOTED_TEXT;
-        preg_match_all("/([^'\"`\[]+)(?:$literal)?/s", $statement, $fragments, PREG_OFFSET_CAPTURE);
+        $quoteCharacterClass = "'";
+        $literal = self::ESCAPED_SINGLE_QUOTED_TEXT . '|';
+        $identifierQuoteCharacter = null !== $platform ? $platform->getIdentifierQuoteCharacter() : null;
+
+        switch ($identifierQuoteCharacter) {
+            case '"':
+                if ($platform instanceof SQLServerPlatform) {
```

Comment:
@maryo I don't think it makes much sense as we cannot profit from it. Applications not knowing that it's SQL Server will assume it is also the ending quote character and still generate invalid SQL. IIRC it is possible to enable ANSI quotes in SQL Server so in that case `"` would work.
Won't open another can here. For now it is only about the platform awareness that's all. Everything else will be addressed in `3.0`